### PR TITLE
Adds useful fields to Subscription and CheckoutSession; fixes bug in CheckoutSession

### PR DESCRIPTION
**Pull request by 781flyingdutchman**
I needed access to a few more fields, so this small change adds:
* to class `Subscription`: `bool cancelAtPeriodEnd`
* to class `CheckoutSession`: `String? status`, `String? url` (the checkout URL)

And fixes an issue where decoding a `CheckoutSession` from JSON was using key `checkout_session` whereas the actual key is `checkout.session`.

I think these would be useful additions to your very useful package. Let me know if you have any questions.

### CODE REVIEW

**Comment by illia-romanenko** [comment]
Hi @781flyingdutchman, thanks for your PR! We landed large change to main - could you please rebase and resolve conflicts?

**Comment by 781flyingdutchman** [comment]
Rebased and resolved conflicts

**Comment by illia-romanenko** [comment]
Nice, thank you!

Could you please also fix Github Actions? 

PS I'll need some way to confirm with the team this  bit as well `checkout_session whereas the actual key is checkout.session.` - if you can provide some context on this change as well (links to documentation, etc.) - that would speed up things. 

Thanks again!

**Comment by 781flyingdutchman** [comment]
I'm not familiar with Actions but I see that you have the Dart SDK pinned to version<3, and that is really old. Because `json_annotations` needed to be upgraded (your version was quite old and it ran into version issues) you need a Dart version 3 or higher, so I think you just need to update the upper end of your Dart SDK range in your pubspec.yaml.

As for the checkout session key: this is how it showed up when I integrated this with Stripe, so I made the change and it works in production for me. I don't have a reference to documentation but I am sure the API returns the dot notation.

**Comment by illia-romanenko** [comment]
> I'm not familiar with Actions but I see that you have the Dart SDK pinned to version<3, and that is really old. Because `json_annotations` needed to be upgraded (your version was quite old and it ran into version issues) you need a Dart version 3 or higher, so I think you just need to update the upper end of your Dart SDK range in your pubspec.yaml.

Feel free to include updated pubspec so we can everything with this PR.

> As for the checkout session key: this is how it showed up when I integrated this with Stripe, so I made the change and it works in production for me. I don't have a reference to documentation but I am sure the API returns the dot notation.

I'll check with the team - it's just weird that all of them are underscore based and this one is dot based. It feels like we might be missing something here.

**Comment by 781flyingdutchman** [comment]
Screenshot 2024-07-02 at 4 36 26 PM

Here is a screenshot from my Stripe dashboard showing the checkout.session event

**Comment by 781flyingdutchman** [comment]
I added the update to pubspec.yaml (set dart sdk to ^3.0.0) but the Action seems to use Dart 2.15 and I'm not sure where it gets that from. As I said, I'm not familiar with Actions so I'm afraid you'll have to solve for that one!

**Comment by illia-romanenko** [comment]
It's located here: https://github.com/enyo/stripe-dart/blob/main/.github/workflows/dart.yml

Do we want to push sdk to 3+ - or is there a way to leave lower constraint as it was? I'm worried that there might be some older/legacy projects?

**Comment by 781flyingdutchman** [comment]
I changed the dart.yml file but it looks like you have to approve it. I don't see a ton of downside to raising the minimum for the Dart SDK to 3.0: if users have older versions then a) the package simply won't update for them (so it doesn't break anything, it just means they cannot enjoy new things) and b) it is very likely that their app is old anyway, as Dart 3 is more than a year old.

**Comment by illia-romanenko** [comment]
Let me check back with the team - to make sure that they are ok with the changes.

File: pubspec.yaml [hunk 1]
```diff
@@ -4,10 +4,11 @@ description: A simple Stripe API wrapper, that is meant to be used on the server
 homepage: https://github.com/enyo/stripe-dart
 documentation: https://github.com/enyo/stripe-dart/blob/main/README.md
 environment:
-  sdk: ">=2.12.0 <3.0.0"
+  sdk: ^3.0.0
 dependencies:
   crypto: ^3.0.0
-  json_annotation: ^4.4.0
+  http: ^0.13.1
```
Comment (illia-romanenko) [inline pubspec.yaml]: Do we need http? As we use dio extensively here.

File: lib/src/messages/checkout_session.dart [hunk 2]
```diff
@@ -52,15 +52,18 @@ class CheckoutSession extends Message {
...
-  });
+  CheckoutSession(
+      {required this.object,
+      required this.id,
+      required this.paymentMethodTypes,
+      this.clientReferenceId,
+      this.customer,
+      this.paymentIntent,
+      this.status,
+      this.url});
```
Comment (illia-romanenko) [inline lib/src/messages/checkout_session.dart]: Could you please add trailing comma to improve formatting?

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
Looks good - almost ready to merge.

File: pubspec.yaml (hunk 1 above)
Comment (781flyingdutchman) [reply pubspec.yaml]: I did not add the http dependency - it was added by Sparks1998* on 3/27 and I merely merged it in. I don't believe there is an actual dependency, so will remove it from the pubspec.yaml

File: lib/src/messages/checkout_session.dart (hunk 2 above)
Comment (781flyingdutchman) [reply lib/src/messages/checkout_session.dart]: Done

**Comment by 781flyingdutchman** [comment]
Made the suggested changes

**Review by illia-romanenko — APPROVED** [review]
LGTM - thank you!

**Comment by 781flyingdutchman** [comment]
Hi, @illia-romanenko,I noticed the package wasn't published with these changes (only the PR merged in the Github repo). Wouldn't it be helpful to publish an update with these additional fields, and keep the package in sync with the main branch?

**Comment by illia-romanenko** [comment]
Hi @781flyingdutchman, unfortunately I have only access to the repo, not to the pub.dev. I hope that @enyo will re-send the invite sometime soon, so I can publish it there.

**Pull request merged**